### PR TITLE
feat(cli): rewrite watch using explicit file globs

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
The goal is to bring multi-stack support to watch, while removing graphql.
In this first iteration we will make it explicit which files we want to watch, ideally we will bring a second iteration where one can specify the files to watch via Constructs.

watch

### CODE REVIEW

File: packages/cdktf-cli/build.ts [hunk 1]
```diff
@@ -50,11 +50,7 @@ const nativeNodeModulesPlugin = {
 
 (async () => {
   await esbuild.build({
-    entryPoints: [
-      "./bin/cdktf.ts",
-      "./lib/server/server.ts",
-      "./bin/cmds/handlers.ts",
-    ],
+    entryPoints: ["./bin/cdktf.ts", "./bin/cmds/handlers.ts"],
```
Comment (DanielMSchmidt) [inline packages/cdktf-cli/build.ts:53]: This one line change has a really weird outcome

### Before the change

```
packages/cdktf-cli/bundle/
├── bin
│   ├── cdktf
│   ├── cdktf.js
│   ├── cdktf.js.map
│   └── cmds
│       ├── handlers.js
│       └── handlers.js.map
├── fsevents-72LCIACT.node
└── lib
    └── server
        ├── schema.graphql
        ├── server.js
        └── server.js.map
```

### After the change

```
packages/cdktf-cli/bundle/
├── bin
│   └── cdktf
├── cdktf.js
├── cdktf.js.map
├── cmds
│   ├── handlers.js
│   └── handlers.js.map
└── lib
    └── server
        └── schema.graphql

4 directories, 6 files
```

Comment (DanielMSchmidt) [reply packages/cdktf-cli/build.ts]: Adding the lib/index.ts (API) helped to get into the old format, we could also change the paths used in the cdktf binary file, but this seemed less intrusive.

Comment (ansgarm) [reply packages/cdktf-cli/build.ts]: Weird 😅 Happy with either solution 😁

File: website/docs/cdktf/create-and-deploy/configuration-file.mdx [hunk 2]
```diff
@@ -119,6 +119,10 @@ Edit `cdktf.json` to disable flags for which you prefer the old behavior, or to
 
 The [CHANGELOG on Github](https://github.com/hashicorp/terraform-cdk/blob/main/CHANGELOG.md) contains a description of new feature flags in each release, and the CDKTF source file [features.ts](https://github.com/hashicorp/terraform-cdk/blob/main/packages/cdktf/lib/features.ts) provides a complete list of all feature flags.
 
+## Configure Files to Watch
+
+When using `cdktf watch` we inspect the `cdktf.json`s `watchPattern` property to determine which files to watch. If you do not specify a `watchPattern` property, we will add the default watch pattern for your language on the first run. The `watchPattern` expects an array of glob patterns, e.g. `["main.ts", "../constructs/**/*.ts", "lib/*.ts"]`.
+
```
Comment (ansgarm) [inline website/docs/cdktf/create-and-deploy/configuration-file.mdx:125]: I really like this!

File: website/docs/cdktf/cli-reference/commands.mdx [hunk 3]
```diff
@@ -460,9 +460,9 @@ $ cdktf synth --app="npm compile && node main.js" --output="dirname"
 
 ~> **Warning:** The `watch` command is experimental, so you should only use it in development environments. It also automatically deploys all changes without asking for confirmation.
 
-The `watch` command watches a directory for changes and automatically synthesizes and deploys changes as they happen. It allows for rapid iterations when developing infrastructure, especially when working with serverless services. It currently supports only one stack at a time and automatically deploys changes without asking for confirmation.
+When the watch command is first run it creates a `watchPattern` in your `cdktf.json` based on the language you configured. It's a list of glob patterns matching all files that should be watched.
```
Comment (ansgarm) [inline website/docs/cdktf/cli-reference/commands.mdx:463]:
```suggestion
When the `watch` command is first run it creates a `watchPattern` in your `cdktf.json` based on the language you configured. It's a list of glob patterns matching all files that should be watched.
```

File: packages/cdktf-cli/package.json [hunk 4]
```diff
@@ -82,10 +82,7 @@
     ]
   },
   "devDependencies": {
-    "@apollo/client": "^3.5.10",
```
Comment (ansgarm) [inline packages/cdktf-cli/package.json:85]: 👋 Byeeeeeee

File: packages/cdktf-cli/lib/watch.ts [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+} from "./cdktf-project";
+import { Errors } from "./errors";
+import * as fs from "fs";
+import * as chokidar from "chokidar";
+import { logger } from "./logging";
+import { CdktfStack } from "./cdktf-stack";
+
+// In this very first iteration we will find out which files to watch by asking the user to provide the files
+// We default to something reasonable per language and write it into the cdkt.json file
+// Mid-Term we might want to add a WatchFile / WatchDir construct that we can use (e.g. in assets) and that a user can use to specify their watch behaviour
```
Comment (ansgarm) [inline packages/cdktf-cli/lib/watch.ts:15]: Do we have follow-up issues for that?

File: packages/cdktf-cli/lib/watch.ts [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        "Watch was in a state where the state was not running in the run phase"
+      );
+    }
+
+    if (state.nextRunQueued) {
+      logger.debug("Next run was queued, starting it now");
+      changeState({
+        ...state,
+        nextRunQueued: false,
+        currentRun: run(),
```
Comment (ansgarm) [inline packages/cdktf-cli/lib/watch.ts:137]: hehe, nice

**Review by ansgarm — APPROVED** [review]
Soo nice 💯 I love that GraphQL is gone now 😁

File: website/docs/cdktf/create-and-deploy/configuration-file.mdx (hunk 2 above)
Comment (DanielMSchmidt) [reply website/docs/cdktf/create-and-deploy/configuration-file.mdx:125]: Yeah, me too :)

Comment (DanielMSchmidt) [reply website/docs/cdktf/create-and-deploy/configuration-file.mdx:125]: Yeah, me too :)

File: packages/cdktf-cli/package.json (hunk 4 above)
Comment (DanielMSchmidt) [reply packages/cdktf-cli/package.json:85]: 👋

File: packages/cdktf-cli/lib/watch.ts (hunk 5 above)
Comment (DanielMSchmidt) [reply packages/cdktf-cli/lib/watch.ts:15]: Created https://github.com/hashicorp/terraform-cdk/issues/1668 for this

**Pull request merged**